spanner/retry.go

```diff
@@ -80,13 +80,13 @@ func (r *spannerRetryer) Retry(err error) (time.Duration, bool) {
 }
 
 // runWithRetryOnAbortedOrSessionNotFound executes the given function and
-// retries it if it returns an Aborted or Session not found error. The retry
-// is delayed if the error was Aborted. The delay between retries is the delay
+// retries it if it returns an Aborted, Session not found error or certain Internal errors. The retry
+// is delayed if the error was Aborted or Internal error. The delay between retries is the delay
 // returned by Cloud Spanner, or if none is returned, the calculated delay with
 // a minimum of 10ms and maximum of 32s. There is no delay before the retry if
 // the error was Session not found.
 func runWithRetryOnAbortedOrSessionNotFound(ctx context.Context, f func(context.Context) error) error {
```

Comment:
Could we have a couple of tests for the changes in this PR that verify that transactions are retried if the `RST_STREAM` error is returned, but not for any other random internal error?

Reply:
@olavloite Test "TestClient_ApplyAtLeastOnce_NonRetryableInternalErrors" added, 
 Please take a look at the code again.
 `This could cause an infinite ` => This is not true. The following code only says that:

if it is an internal error and its error message is not in the list, then return `false`.
If it is an internal error and it error message in the list, then let it pass to the next check `r.Retryer.Retry(err)`.

It early returns `false` for non-retryable internal errors before the real retry check.

If the user has specified to not retry `internal errors`, then `r.Retryer.Retry(err)` will return `false`.